### PR TITLE
chore(tests): added multiple entity metadata tests

### LINKED ISSUE #8042: Document and test that setting metadata to null deletes it
I assumed this would cast to empty string so this surprised me. This is an undocumented quirk of setMetadata's behavior and results in it returning `false`. Effectively a no-op occurs if $multiple is true.

Truthfully I'm not crazy about this behavior, as there's already deleteMetadata() and __unset().

Test would probably go in ElggEntityTest::testElggEntityMetadata().

### PR DESCRIPTION
**Pull request by jdalsem**
Fixes #8042

### CODE REVIEW

**Comment by jdalsem** [comment]
these test need to expose some potential flaws in the current entity setter/getter system

after these tests are fixed i can improve metadata update system to prevent usecases as described in #4268

**Comment by jdalsem** [comment]
ok, some tests are now failing so i am going to fix the following

- [ ] `unset($entity->metadata)` should remove the metadata record
- [ ] setting metadata to an empty array should remove all records thus return null `$entity->md = []`

**Comment by jdalsem** [comment]
> unset($entity->metadata) should remove the metadata record

this was only the case on a unsaved entity

**Comment by jdalsem** [comment]
> setting metadata to an empty array should remove all records thus return null $entity->md = []

this was only an issue on unsaved entities

**Comment by jdalsem** [comment]
ready to be merged

File: engine/tests/phpunit/integration/Elgg/Integration/ElggEntityMetadataTest.php [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+	}
+
+	public function testSetMetadataAppendArrayValueReturnsArray() {
+		foreach ($this->entities as $entity) {
+			$entity->foo = [
+				'brett',
+				'bryan',
+				'brad',
+			];
+			$entity->setMetadata('foo', ['test1', 'test2', 'test3'], '', true);
```
Comment (hypeJunction) [inline engine/tests/phpunit/integration/Elgg/Integration/ElggEntityMetadataTest.php]: I don't get this. Why metadata is merged?

Comment (hypeJunction) [reply engine/tests/phpunit/integration/Elgg/Integration/ElggEntityMetadataTest.php]: I suppose it makes sense, but I never realized you could do this. I don't ever use setMetadata, so I guess I never came across this cork, but honestly doesn't feel very intuitive.

Comment (jdalsem) [reply engine/tests/phpunit/integration/Elgg/Integration/ElggEntityMetadataTest.php]: this was a test that existed and indeed not something that i use also...

**Pull request merged**